app/src/main/java/com/example/vaccinespotter/notifications/VaccineNotificationManager.java

```diff
@@ -4,26 +4,38 @@
 import android.app.NotificationManager;
 import android.content.Context;
 import android.util.Log;
+
 import androidx.core.app.NotificationCompat;
 import androidx.core.app.NotificationManagerCompat;
+
 import com.example.vaccinespotter.R;
 import com.example.vaccinespotter.models.NotificationModel;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
 import java.util.ArrayList;
 import java.util.List;
 
 public class VaccineNotificationManager {
 
+    private final static String FILE_NAME = "data.dat";
     private final static int MAX_CHARACTERS_EXCEPTION_STRING = 100;
     private final String VACCINE_AVAILABLE = "Vaccine available";
     private final String TAG = "VaccineNotificationManager";
-    private int mNotificationId;
     private final Context mContext;
-    private List<NotificationModel> sentNotifications;
+    private int mNotificationId;
+    private List<NotificationModel> mSentNotifications;
     private boolean mRegisteredNotification = false;
 
     public VaccineNotificationManager(Context context) {
         mContext = context;
-        sentNotifications = new ArrayList<>();
+        mSentNotifications = new ArrayList<>();
+        mNotificationId = (int) System.currentTimeMillis();
```

Comment:
We can have time in seconds here, as anyway the notification id is incremented by 1 after this. So no point of collision and also we can avoid such a large value